docroot/modules/humsci/hs_blocks/hs_blocks.module

```diff
@@ -116,3 +116,16 @@ function hs_blocks_preprocess_block__layout_builder(&$variables) {
     $variables['label'] = $variables['configuration']['label'];
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function hs_blocks_preprocess_block__views_exposed_filter_block(&$variables) {
+  if (str_contains($variables['derivative_plugin_id'], "search-search_page")) {
+    // Append the build id to create
+    // unique form id.
+    $build_id = $variables['content']['#build_id'];
+    $variables['content']['#id'] .= '-' . $build_id;
+    $variables['#cache']['max-age'] = 0;
```

Comment:
Are you sure this is needed? I did a quick test with the cache enabled and I still got different ids for both blocks. I might be doing something wrong in my test, but please check to confirm if it's needed.

Reply:
True, even if they get cached the ids will still be unique. You're right, it's not necessary to update the id every time. I'll remove it 👍🏼